packages/metadata/README.md

```diff
@@ -412,6 +412,25 @@ const allParamsForConstructor = MetadataInspector.getAllParameterMetaData<
 >('my-parameter-decorator-key', MyController, '');
 ```
 
+### Use strong-typed metadata access key
+
+You can use MetadataAccessor to provide type checks for metadata access via
+keys. For example,
+
+```ts
+const CLASS_KEY = MetadataAccessor.create<MyClassMetadata, ClassDecorator>(
+    'my-class-decorator-key',
+  );
+
+// Create a class decorator with the key
+const myClassDecorator = ClassDecoratorFactory.createDecorator(CLASS_KEY);
+
+// Inspect a class with the key
+const myClassMeta = MetadataInspector.getClassMetaData(CLASS_KEY, MyController);
```

Comment:
Can we include a comment here insinuating that `myClassMeta` will have its type set to `MyClassMetadata`?